### ISSUE TITLE
Inaccurate version bound

### ISSUE THREAD

**Issue by hvr** [int-index/transformers-lift#7]
As can be seen at https://matrix.hackage.haskell.org/package/transformers-lift@1549836919 `transformers-lift` is not compatible with `transformers >= 0.5.6` due to module name clashes.

I've already revised 

 - https://hackage.haskell.org/package/transformers-lift-0.2.0.0/revisions/
 - https://hackage.haskell.org/package/transformers-lift-0.2.0.1/revisions/

accordingly.

The resulting build reports can be inspected at

https://matrix.hackage.haskell.org/package/transformers-lift@1549837448

**Comment by int-index** [comment]
I checked and I can't find any module name that would clash. What is the exact build failure?

Could it be caused by the `writer-cps-transformers` dependency? The `Control.Monad.Trans.Writer.CPS` module was copied from there.

**Comment by int-index** [comment]
@hvr I figured out how to read the logs, and indeed, the issue is due to `writer-cps-transformers`:

```
src/Control/Monad/Trans/Lift/Local.hs:30:18:
    Ambiguous module name `Control.Monad.Trans.Writer.CPS':
      it was found in multiple packages:
      transformers-0.5.6.2 writer-cps-transformers-0.1.1.4
```

Please undo your changes to my correct version bounds.

**Comment by int-index** [comment]
Okay, come to think of it, if two packages expose equally module names, maybe it is downstream's issue to deal with. I tried to adapt my code to `transformers >= 0.5.6` but it does not expose `.Internal` modules for `Control.Monad.Trans.Writer.CPS` and `Control.Monad.Trans.RWS.CPS`, so that turned out impossible.

**Comment by hvr** [comment]
@int-index hrm... yeah, this one can in theory have multiple ways to be resolved.

But it's not `writer-cps-transformers`'s "fault" that another package suddenly decided to name-clash its modules. In this particular case I believe the best course of action is to use `{-# LANGUAGE PackgeImports #-}` and disambiguate that you indeed want the CPS modules to be taken from `writer-cps-transformers`

**Comment by minad** [comment]
@hvr indeed. The writer-cps-* packages will turn into compatibility packages soon.

**Comment by hvr** [comment]
@minad fyi, recently with the network/network-bsd split, we had a related situation which took some time to figure out and it also shows the trickiness involved (hinted at in the package description): http://hackage.haskell.org/package/network-bsd

**Comment by int-index** [comment]
The thing I can't figure out is how to deal with `.Internal` modules. I need them for some functionality, but `transformers` doesn't seem to expose an unsafe interface to the CPS modules.

**Comment by hvr** [comment]
@int-index if `transformers` doesn't expose things you need, I'm afraid you're out of luck -- unless some dirty `unsafeCoerce` games are played :-)

**Comment by minad** [comment]
Transformers should expose those modules. Can you open an issue please?

**Comment by int-index** [comment]
@minad Alright, I opened an issue there https://hub.darcs.net/ross/transformers/issue/67
